test/core/event_engine/posix/posix_endpoint_test.cc

```diff
@@ -0,0 +1,352 @@
+    // been deleted.
+    Unref();
+  }
+  std::shared_ptr<EventEngine> engine_;
+  PosixEventPoller* poller_;
+  grpc_core::Notification signal;
+};
+
+class PosixEndpointTest : public ::testing::TestWithParam<TestParam> {
+  void SetUp() override {
+    posix_ee_ = std::make_shared<PosixEventEngine>();
+    scheduler_ =
+        absl::make_unique<grpc_event_engine::posix_engine::TestScheduler>(
+            posix_ee_.get());
+    EXPECT_NE(scheduler_, nullptr);
+    GPR_GLOBAL_CONFIG_SET(grpc_poll_strategy, GetParam().Poller().c_str());
+    poller_ = GetDefaultPoller(scheduler_.get());
+    if (poller_ != nullptr) {
+      EXPECT_EQ(poller_->Name(), GetParam().Poller());
+    }
+  }
+
+  void TearDown() override {
+    if (poller_ != nullptr) {
+      poller_->Shutdown();
+    }
+    WaitForPendingTasks(std::move(posix_ee_));
+  }
+
+ public:
+  TestScheduler* Scheduler() { return scheduler_.get(); }
+
+  std::shared_ptr<EventEngine> GetPosixEE() { return posix_ee_; }
+
+  PosixEventPoller* PosixPoller() { return poller_; }
+
+ private:
+  PosixEventPoller* poller_;
+  std::unique_ptr<TestScheduler> scheduler_;
+  std::shared_ptr<EventEngine> posix_ee_;
+};
+
+TEST_P(PosixEndpointTest, ConnectExchangeBidiDataTransferTest) {
+  if (PosixPoller() == nullptr) {
+    return;
+  }
+  Worker* worker = new Worker(GetPosixEE(), PosixPoller());
+  worker->Start();
+  {
+    auto connections = CreateConnectedEndpoints(
+        PosixPoller(), GetParam().IsZeroCopyEnabled(), 1, GetPosixEE());
+    auto it = connections.begin();
+    auto client_endpoint = std::move(std::get<0>(*it));
+    auto server_endpoint = std::move(std::get<1>(*it));
+    EXPECT_TRUE(client_endpoint != nullptr);
+    EXPECT_TRUE(server_endpoint != nullptr);
+    connections.erase(it);
+
+    // Alternate message exchanges between client -- server and server --
+    // client.
+    for (int i = 0; i < kNumExchangedMessages; i++) {
+      // Send from client to server and verify data read at the server.
+      ASSERT_TRUE(SendValidatePayload(GetNextSendMessage(),
+                                      client_endpoint.get(),
+                                      server_endpoint.get())
+                      .ok());
+      // Send from server to client and verify data read at the client.
+      ASSERT_TRUE(SendValidatePayload(GetNextSendMessage(),
+                                      server_endpoint.get(),
+                                      client_endpoint.get())
+                      .ok());
+    }
+  }
+  worker->Wait();
+}
+
+// Create  N connections and exchange and verify random number of messages over
+// each connection in parallel.
+TEST_P(PosixEndpointTest, MultipleIPv6ConnectionsToOneOracleListenerTest) {
+  if (PosixPoller() == nullptr) {
+    return;
+  }
+  Worker* worker = new Worker(GetPosixEE(), PosixPoller());
+  worker->Start();
+  auto connections =
+      CreateConnectedEndpoints(PosixPoller(), GetParam().IsZeroCopyEnabled(),
```

Comment:
Calling PosixPoller again creates a new instance, right?

Reply:
No it just returns the same pointer. There is one poller instructed per test.